src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -130,11 +130,16 @@ const connectInfiniteHits: InfiniteHitsConnector = (
         // if we only change pages.
         const {
           page = 0,
+          facets,
           hierarchicalFacets,
           disjunctiveFacets,
+          maxValuesPerFacet,
```

Comment:
This is very brittle each time a widget that deals with parameters is added we have to check if we have to ignore it or not. Even a user that manipulate the state could end up with an unwanted reset page. I don't have a solution be we should find one at some point. We could release with this solution but IMO it has to be temporary.

Reply:
Agree, I said in https://github.com/algolia/instantsearch.js/pull/4104 that it was very hacky. We'll think of a clean solution after the release.